File: javascript/src/getToken.js
import dotenv from 'dotenv';

dotenv.config();

const BASE_URL = 'https://api.hyperswitchai.com';

async function main() {
  try {

    if (!process.env.USERNAME || !process.env.PASSWORD) {
      throw new Error('USERNAME and PASSWORD must be set in a .env file in the javascript directory. You can rename .env.sample to .env and fill in your own email and password.');
    }
    
    const authResponse = await fetch(`${BASE_URL}/auth`, {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({
          username: process.env.USERNAME,
          password: process.env.PASSWORD,
        })
      });
  
      if (!authResponse.ok) {
        throw new Error('Authentication failed');
      }
  
      const { token } = await authResponse.json();

      console.log('\n\x1b[32m%s\x1b[0m', '✅ Token successfully retrieved');
      console.log('Token:', token);

  } catch (error) {
    console.error('Error:', error);
  }
}

main();

File: ruby/src/get_token.rb
require 'dotenv'
require 'faraday'
require 'json'

# Load environment variables from .env file
Dotenv.load

BASE_URL = 'https://api.hyperswitchai.com'

def main
  # Check for required environment variables
  unless ENV['USERNAME'] && ENV['PASSWORD']
    raise 'USERNAME and PASSWORD must be set in a .env file in the ruby directory. You can rename .env.sample to .env and fill in your own email and password.'
  end

  # Create HTTP client
  conn = Faraday.new(url: BASE_URL) do |f|
    f.request :json # Encode request bodies as JSON
    f.response :json # Decode response bodies as JSON
  end

  begin
    # Make authentication request
    response = conn.post('/auth') do |req|
      req.headers['Content-Type'] = 'application/json'
      req.body = {
        username: ENV['USERNAME'],
        password: ENV['PASSWORD']
      }
    end

    # Check response status
    unless response.success?
      raise 'Authentication failed'
    end

    # Extract token from response
    token = response.body['token']

    puts "\n\e[32m✅ Token successfully retrieved\e[0m"
    puts "Token: #{token}"

  rescue => e
    puts "Error: #{e.message}"
  end
end

# Run if this file is executed directly
main if __FILE__ == $PROGRAM_NAME 